unit_tests/log_transfer/test_main_chkpt_info.cpp

```diff
@@ -0,0 +1,800 @@
+  tdes->has_deadlock_priority = false;
+
+  tdes->num_log_records_written = 0;
+
+  LSA_SET_NULL (&tdes->rcv.tran_start_postpone_lsa);
+  LSA_SET_NULL (&tdes->rcv.sysop_start_postpone_lsa);
+  LSA_SET_NULL (&tdes->rcv.atomic_sysop_start_lsa);
+  LSA_SET_NULL (&tdes->rcv.analysis_last_aborted_sysop_lsa);
+  LSA_SET_NULL (&tdes->rcv.analysis_last_aborted_sysop_start_lsa);
+}
+
+log_tdes *
+systdes_create_tdes ()
+{
+  log_tdes *tdes = new log_tdes ();
+  return tdes;
+}
+
+log_tdes *
+log_system_tdes::rv_get_tdes (TRANID trid)
+{
+  auto it = systb_System_tdes.find (trid);
+  if (it != systb_System_tdes.end ())
+    {
+      return it->second;
+    }
+  else
+    {
+      return NULL;
+    }
+}
+
+LOG_TDES *
+logtb_rv_find_allocate_tran_index (THREAD_ENTRY *thread_p, TRANID trid, const LOG_LSA *log_lsa)
+{
+  LOG_TDES *tdes = NULL;		/* Transaction descriptor */
+
+  assert (trid != NULL_TRANID);
+
+  for (int i = 1; i < NUM_TOTAL_TRAN_INDICES; i++)
+    {
+      tdes = log_Gl.trantable.all_tdes[i];
+      if (tdes != NULL && tdes->trid != NULL_TRANID && tdes->trid == trid)
+	{
+	  LSA_COPY (&tdes->head_lsa, log_lsa);
+	  tran_map.insert (std::pair<TRANID, LOG_TDES *> (trid, tdes));
+	  return tdes;
+	}
+    }
+  assert (false);
+  return tdes;
+}
+
+void
+logpb_fatal_error (THREAD_ENTRY *thread_p, bool log_exit, const char *file_name, const int lineno, const char *fmt,
+		   ...)
+{
+  assert (false);
+}
+
+static const int LOG_TOPOPS_STACK_INCREMENT = 3;	/* No more than 3 nested top system operations */
+void *
+logtb_realloc_topops_stack (LOG_TDES *tdes, int num_elms)
+{
+  size_t size;
+  void *newptr;
+
+  if (num_elms < LOG_TOPOPS_STACK_INCREMENT)
+    {
+      num_elms = LOG_TOPOPS_STACK_INCREMENT;
+    }
+
+  size = tdes->topops.max + num_elms;
+  size = size * sizeof (*tdes->topops.stack);
+
+  newptr = (LOG_TOPOPS_ADDRESSES *) realloc (tdes->topops.stack, size);
+  if (newptr != NULL)
+    {
+      tdes->topops.stack = (LOG_TOPOPS_ADDRESSES *) newptr;
+      if (tdes->topops.max == 0)
+	{
+	  tdes->topops.last = -1;
+	}
+      tdes->topops.max += num_elms;
+    }
+  else
+    {
+      return NULL;
+    }
+  return tdes->topops.stack;
+}
+
+PGLENGTH
+db_log_page_size (void)
+{
+  return 16384;
+}
+
+int
+logtb_get_current_tran_index (void)
+{
+  return 1;
+}
+
+int
+log_read_sysop_start_postpone (THREAD_ENTRY *thread_p, LOG_LSA *log_lsa, LOG_PAGE *log_page, bool with_undo_data,
+			       LOG_REC_SYSOP_START_POSTPONE *sysop_start_postpone, int *undo_buffer_size,
+			       char **undo_buffer, int *undo_size, char **undo_data)
+{
+  assert (true);
+  return NO_ERROR;
+}
+
+int
+logtb_find_tran_index (THREAD_ENTRY *thread_p, TRANID trid)
+{
+  int i;
+  int tran_index = NULL_TRAN_INDEX;	/* The transaction index */
+  LOG_TDES *tdes;		/* Transaction descriptor */
+
+  assert (trid != NULL_TRANID);
+
+  /* Avoid searching as much as possible */
+  tran_index = LOG_FIND_THREAD_TRAN_INDEX (thread_p);
+  tdes = LOG_FIND_TDES (tran_index);
```

Comment:
This is an optimization that works on the server, but it will not help for the unit test. It is better to search directly in the transaction table.